### PR TITLE
Make sure the cache for postings cardinality properly honors the label name

### PR DESCRIPTION
**Pull request by damnever**

### CODE REVIEW

**Comment by bboreham** [comment]
Hi, thanks for the PR, I think I see what you had in mind.
Do you think you could add a test which fails in the 'before' case?

**Comment by damnever** [comment]
Tests added.

**Comment by damnever** [comment]
@jesusvazquez Please take a look.

File: tsdb/head_test.go [hunk 1]
```diff
@@ -5399,3 +5399,26 @@ func TestCuttingNewHeadChunks(t *testing.T) {
...
 	}
 }
+
+func TestPostingsCardinalityStats(t *testing.T) {
+	head := &Head{postings: index.NewMemPostings()}
+	head.postings.Add(1, labels.FromStrings(labels.MetricName, "t", "n", "v1"))
+	head.postings.Add(2, labels.FromStrings(labels.MetricName, "t", "n", "v2"))
+
+	statsForMetricName := head.PostingsCardinalityStats(labels.MetricName, 10)
+	// Using cache.
```
Comment (bboreham) [inline tsdb/head_test.go:5409]: What does this comment refer to?

**Review by bboreham — APPROVED** [review]
Thanks, seems fine.

File: tsdb/head_test.go (hunk 1 above)
Comment (damnever) [reply tsdb/head_test.go]: This means that the cache is used within the valid duration, and adding a new series does not change the result. I have moved this down by one line.

**Comment by bwplotka** [comment]
(During bug scrub).

Sorry for the delay. Sounds like we still need the DCO check to pass (please sign the commit) and we might need a rebase, otherwise it should be good to go!

**Comment by bboreham** [comment]
Lint errors should be addressed by https://github.com/prometheus/prometheus/pull/13175

**Review by bboreham — APPROVED** [review]
Thanks!

**Pull request merged**